30147 kalimaster jpo

KALjpo opened the pull request.

Commit:
Ajout fichier specs

Commit:
Specs

ydomenjoud commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-button/kal-button.component.spec.ts:
```diff
@@ -1,25 +1,81 @@
+import { By } from '@angular/platform-browser';
 
-describe('KalButtonComponent', () => {
+fdescribe('KalButtonComponent', () => {
```
remove 'f'

ydomenjoud commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-button/kal-button.component.ts:
```diff
@@ -9,9 +13,48 @@ import { Component, OnInit, ViewEncapsulation, ChangeDetectionStrategy } from '@
+  /**
+   * Is the button disabled ?
+   */
+  @Input()
```
Annotation @Input on setter, not getter

Commit:
https://github.com/kalidea/kaligraphi/pull/2#discussion_r226436763
https://github.com/kalidea/kaligraphi/pull/2#discussion_r226436547

jsp-up commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-button/kal-button.component.spec.ts:
```diff
@@ -1,25 +1,81 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { KalButtonComponent } from './kal-button.component';
+import { By } from '@angular/platform-browser';
```
L'import doit se situer en dessous du premier @angular

jsp-up commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-button/kal-button.component.spec.ts:
```diff
@@ -1,25 +1,81 @@
     fixture = TestBed.createComponent(KalButtonComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
+    // fixture.detectChanges();
```
supprimer le commentaire

jsp-up commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-button/kal-button.component.spec.ts:
```diff
@@ -1,25 +1,81 @@
     expect(component).toBeTruthy();
   });
+
+  it('should emit event on clic', () => {
```
click

jsp-up commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-button/kal-button.component.spec.ts:
```diff
@@ -1,25 +1,81 @@
+  });
+
+  it('should create tabIndex', () => {
+    const toto = 5;
```
renommer la variable

jsp-up commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-button/kal-button.component.ts:
```diff
@@ -9,7 +13,46 @@ import { Component, OnInit, ViewEncapsulation, ChangeDetectionStrategy } from '@
 
-  constructor() { }
+  /**
+   * Is the button labelised ?
```
Button label

jsp-up commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-button/kal-button.component.ts:
```diff
@@ -9,7 +13,46 @@ import { Component, OnInit, ViewEncapsulation, ChangeDetectionStrategy } from '@
+
+  set disabled(value: boolean) {
+    this.isDisabled = coerceBooleanProperty(value);
+    this.tabIndex = this.isDisabled ? this.tabIndex : null;
```
J'aurais réutilisé le getter :
`this.tabIndex = this.disabled ? this.tabIndex : null;`

KALjpo replied:
??

Commit:
https://github.com/kalidea/kaligraphi/pull/2#discussion_r226586284
https://github.com/kalidea/kaligraphi/pull/2#discussion_r226586460
https://github.com/kalidea/kaligraphi/pull/2#discussion_r22658646
https://github.com/kalidea/kaligraphi/pull/2#discussion_r226586491
https://github.com/kalidea/kaligraphi/pull/2#discussion_r226586763
https://github.com/kalidea/kaligraphi/pull/2#discussion_r226586940

Commit:
cas icone

Commit:
Merge branch 'master' into 30147_KALIMASTER_JPO

Commit:
cas icone

Commit:
cas icone

Commit:
Merge branch 'master' into 30147_KALIMASTER_JPO

Commit:
Merge branch 'master' into 30147_KALIMASTER_JPO

Commit:
cas icone

Commit:
cas icone

Commit:
Fix variable
